### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by elarhadu**
In this milestone;
- [x] I added a column species of type string to my `animals` table.

- [x] In a transaction, I updated the animals table by setting the species column to unspecified. Then verified that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction
query 9

- [x] I updated the animals table by setting the species column to digimon for all animals that have a name ending in mon and also updated the animals table by setting the species column to pokemon for all animals that don't have species already set.
query 10
query 11

- [x] Inside a transaction, I deleted all records in the `animals` table, then roll back the transaction.
query 12

- [x] Delete all animals born after Jan 1st, 2022.
query 13

- [x] I created a savepoint for the transaction. Update all animals' weight to be their weight multiplied by -1. Rollback to the savepoint
query 14

- [x] Update all animals' weights that are negative to be their weight multiplied by -1 and Commit transaction

query 15

- [x] How many animals are there? How many animals have never tried to escape? What is the average weight of animals? Who escapes the most, neutered or not neutered animals?

query 16-19

- [x] What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
query 20-23

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -5,4 +5,22 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon','Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered= 'TRUE';
 SELECT * FROM animals WHERE NOT name='Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- TRANSACTIONS
```
Comment (Ghiftee) [inline queries.sql:10]: - [ ] The first query should be UPDATE the animals table by setting the species column to unspecified, then roll back the change. eg
```sql
BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,22 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon','Pikachu');
...
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
```
Comment (Ghiftee) [inline queries.sql:26]: - [ ] After this query, according to the instructions, kindly commit the transaction and verify that the change was made e.g.
```sql
SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 3]
```diff
@@ -5,4 +5,22 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon','Pikachu');
...
+DELETE FROM animals;
+DELETE FROM animals WHERE date_of_birth>'2022-01-01';
```
Comment (Ghiftee) [inline queries.sql:48]: - [ ] After deleting all animals, kindly roll back the transaction and ensure that it was successful e.g
```sql
ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

File: queries.sql [hunk 4]
```diff
@@ -5,4 +5,22 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon','Pikachu');
...
+SAVEPOINT SPGABUMON;
+UPDATE animals SET weight_kg=weight_kg * -1;
```
Comment (Ghiftee) [inline queries.sql:50]: - [ ] After this query, kindly ensure you rollback to the savepoint.

File: queries.sql [hunk 5]
```diff
@@ -5,4 +5,22 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon','Pikachu');
...
+UPDATE animals SET weight_kg=weight_kg * -1 WHERE weight_kg<0;
+
```
Comment (Ghiftee) [inline queries.sql:53]: - [ ] Let's not forget to commit the transaction here.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @elarhadu 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Query screenshots are added
✔️ Required data is added
✔️ README is professional

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._
- [ ] Good job adding all the commits and rollbacks in your screenshot, however, the queries still need to be written. Kindly add them as highlighted below.
- [ ] Kindly ensure your MIT link is fixed as this would give your README a more professional look.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**